exercises/harriet_brown/ex02_oo_basics/src/main.cpp

```diff
@@ -0,0 +1,69 @@
+/**
+ * Skeleton main routine
+ */
+#include <iostream>
+#include <vector>
+#include <string>
+#include "../inc/ShapeSorter.h"
+#include "../inc/getInput.h"
+
+
+void setup_new_shape(std::vector<Shape*> &shapes)
+{
+	std::string shape_type;
+	double side1;
+	double side2;
+	getInput("Shape: ", shape_type);
+	if (shape_type == "square") {
+		getInput("Side size: ", side1);
+		shapes.push_back(new Square(side1));
```

Comment:
These days we would try to use a [`std::unique_ptr`](https://en.cppreference.com/w/cpp/memory/unique_ptr) to avoid having to call an explicit `new` as this avoids a memory leak. So here we would have `std::vector<std::unique_ptr<Shape>>` and then `shapes.push_back(std::make_unique<Square>(side1))` for example.